### PR TITLE
0.1% accuracy

### PR DESCRIPTION
**Pull request by xzackli**
This PR includes modifications to the hierarchy which allows Bolt transfer functions to reach 0.1% agreement with CLASS in perturbations like Phi, delta_c, delta_b, and the photon monopole. 

This PR also includes a change of units! Previously we were using natural units with a base unit of eV. This is not very useful for cosmology -- we join the rest of the world and use Megaparsecs as the base natural unit.

### CODE REVIEW

File: scripts/plot_perts_x_nonu.jl [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#['k (h/Mpc)', 'd_g', 'd_b', 'd_cdm', 'd_ur', 'd_ncdm[0]', 'd_tot',
+#'phi', 'psi', 't_g', 't_b', 't_cdm', 't_ur', 't_ncdm[0]', 't_tot']
+retnf = open( @sprintf("./test/data/zack_N_class_px_k%s_nofluid_nonu.dat",k_choice),"r" ) do datafile
+# an example that goes to early times -> retnf = open("./test/data/lowres_class_px_kp03_nofluid.dat","r") do datafile
+    [parse.(Float64, split(line)) for line in eachline(datafile)]
+end
+#the second column is just a repeated k value, so remember it and delete col
+kclass = retnf[2][1] #read class k mode from file (in h/Mpc)
+# k = (bg.H₀*3e5/100)*kclass #get k in our units ->old value
+k = 𝕡.h*kclass  #get k in our units
```
Comment (xzackli) [inline scripts/plot_perts_x_nonu.jl:34]: Code is now in Mpc units, so the wavenumber are in 1/Mpc.

File: src/Bolt.jl [hunk 2]
```diff
@@ -26,23 +27,34 @@ import SpecialFunctions: lgamma, sphericalbesselj
...
 import PhysicalConstants.CODATA2018: ElectronMass, ProtonMass,
     FineStructureConstant, ThomsonCrossSection, NewtonianConstantOfGravitation
 
 
+# set the unit system, this should be improved
+natural(x) = UnitfulCosmo.mpc(x)
+unnatural(x, y) = UnitfulCosmo.unmpc(x, y)
+# natural(x) = NaturallyUnitful.natural(x)
+# unnatural(x, y) = NaturallyUnitful.unnatural(x, y)
+
```
Comment (xzackli) [inline src/Bolt.jl:41]: This is the main change to go from eV to Mpc base units.

File: src/perturbations.jl [hunk 3]
```diff
@@ -259,8 +267,8 @@ function hierarchy!(du, u, hierarchy::Hierarchy{T, BasicNewtonian}, x) where T
         end
 
         # photon boundary conditions: diffusion damping
-        Θ′[ℓᵧ] = k / ℋₓ * Θ[ℓᵧ-1] - (ℓᵧ + 1) / (ℋₓ * ηₓ) + τₓ′ * Θ[ℓᵧ]
-        Θᵖ′[ℓᵧ] = k / ℋₓ * Θᵖ[ℓᵧ-1] - (ℓᵧ + 1) / (ℋₓ * ηₓ) + τₓ′ * Θᵖ[ℓᵧ]
+        Θ′[ℓᵧ] = k / ℋₓ * Θ[ℓᵧ-1] - ( (ℓᵧ + 1) / (ℋₓ * ηₓ) - τₓ′ ) * Θ[ℓᵧ]
+        Θᵖ′[ℓᵧ] = k / ℋₓ * Θᵖ[ℓᵧ-1] - ( (ℓᵧ + 1) / (ℋₓ * ηₓ) - τₓ′ ) * Θᵖ[ℓᵧ]
```
Comment (xzackli) [inline src/perturbations.jl:271]: New truncation from MB via Jamie

File: src/perturbations.jl [hunk 4]
```diff
@@ -288,9 +296,10 @@ function initial_conditions(xᵢ, hierarchy::Hierarchy{T, BasicNewtonian}) where
     # ρ0ℳ = bg.ρ₀ℳ(xᵢ)
 
     # metric and matter perturbations
-    Φ = 1.0
+    ℛ = 1.0  # set curvature perturbation to 1
+    Φ = (4f_ν + 10) / (4f_ν + 15) * ℛ  # for a mode outside the horizon in radiation era
     #choosing Φ=1 forces the following value for C, the rest of the ICs follow
-    C = -( (15 + 4f_ν)/(20 + 8f_ν) )
+    C = -( (15 + 4f_ν)/(20 + 8f_ν) ) * Φ
```
Comment (xzackli) [inline src/perturbations.jl:302]: Now normalized to curvature = 1

File: src/background.jl [hunk 5]
```diff
@@ -78,7 +75,7 @@ H(x, par::AbstractCosmoParams,quad_pts,quad_wts) = H_a(x2a(x),par,quad_pts,quad_
 # conformal time
 function η(x, par::AbstractCosmoParams,quad_pts,quad_wts)
     #fast copy from q - need to check accuracy (#FIXME) but a plays the role of q
-    logamin,logamax=-11.75,log10(x2a(x)) #0,x2a(x)
+    logamin,logamax=-13.75,log10(x2a(x)) #0,x2a(x)
```
Comment (xzackli) [inline src/background.jl:78]: background integration bug fixed by jamie!

File: src/perturbations.jl [hunk 6]
```diff
@@ -288,9 +296,10 @@ function initial_conditions(xᵢ, hierarchy::Hierarchy{T, BasicNewtonian}) where
     # ρ0ℳ = bg.ρ₀ℳ(xᵢ)
 
     # metric and matter perturbations
-    Φ = 1.0
+    ℛ = 1.0  # set curvature perturbation to 1
+    Φ = (4f_ν + 10) / (4f_ν + 15) * ℛ  # for a mode outside the horizon in radiation era
```
Comment (xzackli) [inline src/perturbations.jl:300]: should check this, seems to work

**Comment by marius311** [comment]
Docs now build Jamie's notebook, see preview here: https://xzackli.github.io/Bolt.jl/previews/PR69/basic_usage/

**Comment by marius311** [comment]
Fwiw @xzackli if you're annoyed by test deps, I would recommend switching to having a `test/Project.toml` for test dependencies, rather than putting them in the main Project.toml, I think that's the current recommended thing. An added bonus is you can activate the testing environment with `using TestEnv; TestEnv.activate()` and you'll get the exact environment the tests see. I don't think the current thing works with that.

**Comment by xzackli** [comment]
Wow -- I hated the way I had to use test deps, this is a nice development. I'll change it over.

**Pull request merged**